cargo-dist/src/linkage.rs

```diff
@@ -434,3 +437,57 @@ pub fn determine_linkage(path: &Utf8PathBuf, target: &str) -> DistResult<Linkage
 
     Ok(linkage)
 }
+
+/// Determine the build environment on the current host
+/// This should be done local to the builder!
+pub fn determine_build_environment(target: &str) -> DistResult<BuildEnvironment> {
+    if target.contains("darwin") {
+        determine_macos_build_environment()
+    } else if target.contains("linux") {
+        determine_linux_build_environment()
+    } else if target.contains("windows") {
```

Comment:
Hmm errors here are hard... this is conceptually very similar to linkage which we don't consider an error to fail to collect. I think it would be correct to do the same here? I don't want to fail someone's release process on this, even if it means the installer doesn't have perfect information.

Reply:
Ensured this returns an indeterminate value instead of an error if we failed to collect information or somehow ran cross-platform.